core/src/com/unciv/ui/pickerscreens/FoundReligionPickerScreen.kt

```diff
@@ -101,36 +95,53 @@ class FoundReligionPickerScreen (
+                ImageGetter.getCircledReligionIcon(religionName, 60f),
+                skin
+            )
+            button.disable()
+            iconsTable.add(button).pad(5f)            
+        }
+        topReligionIcons.add(iconsTable).padBottom(10f).row()
+        topReligionIcons.add(descriptionLabel).center().padBottom(5f)
+    }
+
     private fun updateLeftTable() {
         leftChosenBeliefs.clear()
-        val currentReligion = choosingCiv.religionManager.religion ?: Religion("Unknown", gameInfo, choosingCiv.civName)
+        val currentReligion = choosingCiv.religionManager.religion ?: Religion("None", gameInfo, choosingCiv.civName)
         
-        for (pantheon in currentReligion.getPantheonBeliefs() + currentReligion.getFollowerBeliefs()) {
-            val beliefButton = convertBeliefToButton(pantheon)
+        for (belief in 
+            currentReligion.getPantheonBeliefs() 
+            + currentReligion.getFounderBeliefs() 
+            + currentReligion.getFollowerBeliefs() 
+            + currentReligion.getEnhancerBeliefs()
+        ) {
+            val beliefButton = convertBeliefToButton(belief)
             leftChosenBeliefs.add(beliefButton).pad(10f).row()
             beliefButton.disable()
         }
+
+        for (newPantheonBelief in beliefsContainer.chosenPantheonBeliefs.withIndex()) {
+            addChoosableBeliefButton(newPantheonBelief, BeliefType.Pantheon)
+        }
         
-        for (newFollowerBelief in chosenFollowerBeliefs.withIndex()) {
-            val newFollowerBeliefButton =
-                if (newFollowerBelief.value == null) emptyBeliefButton(BeliefType.Follower)
-                else convertBeliefToButton(newFollowerBelief.value!!)
-                
-            leftChosenBeliefs.add(newFollowerBeliefButton).pad(10f).row()
-            newFollowerBeliefButton.onClick {
-                loadRightTable(BeliefType.Follower, newFollowerBelief.index)
-            }
+        for (newFounderBelief in beliefsContainer.chosenFounderBeliefs.withIndex()) {
+            addChoosableBeliefButton(newFounderBelief, BeliefType.Founder)
         }
         
-        for (newFounderBelief in chosenFounderBeliefs.withIndex()) {
-            val newFounderBeliefButton =
-                if (newFounderBelief.value == null) emptyBeliefButton(BeliefType.Founder)
-                else convertBeliefToButton(newFounderBelief.value!!)
-
-            leftChosenBeliefs.add(newFounderBeliefButton).pad(10f).row()
-            newFounderBeliefButton.onClick {
-                loadRightTable(BeliefType.Founder, newFounderBelief.index)
-            }
+        for (newFollowerBelief in beliefsContainer.chosenFollowerBeliefs.withIndex()) {
+            addChoosableBeliefButton(newFollowerBelief, BeliefType.Follower)
+        }
+        
+        for (newEnhancerBelief in beliefsContainer.chosenEnhancerBeliefs.withIndex()) {
+            addChoosableBeliefButton(newEnhancerBelief, BeliefType.Enhancer)
```

Comment:
This, and BeliefContainer, are way overcomplicated
Just save the Beliefs as-is - no index, in an ArrayList<chosenBeliefs>, and filter when needed
You'll save like 40 lines of code and it'll be much easier to read :)